### PR TITLE
114 add e2e tests for builtin commands

### LINKED ISSUE #114: Add e2e tests for builtin commands

### PR DESCRIPTION
**Pull request by alex216**
fix #114 

## Release Notes

- **New Features**
  - Enhanced error handling for the `cd` command, including checks for the `HOME` environment variable.
  - Improved path normalization to handle consecutive slashes.

- **Bug Fixes**
  - Streamlined error reporting for environment variable exports, ensuring accurate error messages.

- **Tests**
  - Added comprehensive end-to-end tests for the `cd`, `export`, `unset`, and `exit` commands, covering various edge cases and error scenarios.

### CODE REVIEW

File: tests/e2e/test_exit.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    shell_session.sendline("exit 298")
+    shell_session.expect("exit")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("echo $?")
+    shell_session.expect(PROMPT)
+    result = get_command_output(shell_session.before)
+    assert result == "42"  # 298 % 256 = 42
+
+
+def test_exit_with_mixed_valid_invalid(shell_session):
```
Comment (rask24) [inline tests/e2e/test_exit.py:140]: same above

File: tests/e2e/test_export.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+def test_export_no_value(shell_session):
+    shell_session.sendline("./minishell")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("export TEST_EMPTY")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("export | grep TEST_EMPTY")
+    shell_session.expect(PROMPT)
+    result = get_command_output(shell_session.before)
+    assert "declare -x TEST_EMPTY" in result
```
Comment (rask24) [inline tests/e2e/test_export.py:36]:
``` suggestion
    assert result == "declare -x TEST_EMPTY"
```

File: tests/e2e/test_export.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+def test_export_no_key(shell_session):
+    shell_session.sendline("./minishell")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("export =value")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("echo $?")
+    shell_session.expect(PROMPT)
+    result = get_command_output(shell_session.before)
+    assert result == "1"
```
Comment (rask24) [inline tests/e2e/test_export.py:61]:
``` suggestion
def test_export_no_key(shell_session):
    shell_session.sendline("./minishell")
    shell_session.expect(PROMPT)
    shell_session.sendline("export =value")
    shell_session.expect(PROMPT)

    result = get_command_output(shell_session.before)
    assert result == "minishell: export: `=value': not a valid identifier"

    shell_session.sendline("echo $?")
    shell_session.expect(PROMPT)
    result = get_command_output(shell_session.before)
    assert result == "1"
```

File: tests/e2e/test_export.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+def test_export_invalid_chars(shell_session):
+    shell_session.sendline("./minishell")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("export TEST@VAR=invalid")
+    shell_session.expect(PROMPT)
+    shell_session.sendline("echo $?")
+    shell_session.expect(PROMPT)
+    result = get_command_output(shell_session.before)
+    assert result == "1"
```
Comment (rask24) [inline tests/e2e/test_export.py:141]:
``` suggestion
def test_export_invalid_chars(shell_session):
    shell_session.sendline("./minishell")
    shell_session.expect(PROMPT)
    shell_session.sendline("export TEST@VAR=invalid")
    shell_session.expect(PROMPT)

    result = get_command_output(shell_session.before)
    assert result == "minishell: export: `TEST@VAR=invalid': not a valid identifier"

    shell_session.sendline("echo $?")
    shell_session.expect(PROMPT)
    result = get_command_output(shell_session.before)
    assert result == "1"
```

File: tests/e2e/test_unset.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        shell_session.expect(PROMPT)
+
+        shell_session.sendline("echo $PATH")
+        shell_session.expect(PROMPT)
+        result = get_command_output(shell_session.before)
+        assert result == ""
+
+    finally:
+        shell_session.sendline(f"export PATH='{original_path}'")
+        shell_session.expect(PROMPT)
```
Comment (rask24) [inline tests/e2e/test_unset.py:61]: I think restoring PATH is not needed. Because each test forks minishell on execution.

**Review by rask24 — COMMENTED** [review]
@alex216 Please check the comments!

**Review by rask24 — APPROVED** [review]
@alex216 LGTM 🏃🏼‍♂️
